cmd/tsdb/main.go

```diff
@@ -90,6 +94,17 @@ func main() {
 			exitWithError(fmt.Errorf("Block not found"))
 		}
 		analyzeBlock(block, *analyzeLimit)
+	case compactCmd.FullCommand():
+		db, err := tsdb.Open(*compactPath, nil, nil, nil)
+		defer db.Close()
```

Comment:
This should come after the error check below, as `db` can be `nil` when there is error. (same for the other 2 places above)